### PR TITLE
Documented how to create lazy-loaded Twig extensions

### LINKED ISSUE #7169: Add docs for the new decoupled Twig extensions
Thanks to some new Twig features, you can now implement the functions/filters/tests outside of the class that defines them. See https://github.com/symfony/symfony/pull/20093, https://github.com/symfony/symfony/pull/20094 and https://github.com/symfony/symfony/pull/20092.

Should we document or mention this?

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #7169.

### CODE REVIEW

File: templating/twig_extension.rst [hunk 1]
```diff
@@ -58,14 +58,114 @@ Create a class that extends ``\Twig_Extension`` and fill in the logic::
...
 Next, register your class as a service and tag it with ``twig.extension``. If you're
 using the :ref:`default services.yml configuration <service-container-services-load-example>`,
 you're done! Symfony will automatically know about your new service and add the tag.
 
 You can now start using your filter in any Twig template.
 
+Creating Lazy-Loaded Twig Extensions
+------------------------------------
+
+.. versionadded:: 1.26
```
Comment (fabpot) [inline templating/twig_extension.rst]: Don't we refer to Symfony versions here? Having a Twig version seems weird to me.

File: templating/twig_extension.rst [hunk 2]
```diff
@@ -58,14 +58,114 @@ Create a class that extends ``\Twig_Extension`` and fill in the logic::
...
+Then, create the new ``AppRuntime`` class (it's not required but these classes
+are suffixed with ``Runtime`` by convention) and include the logic of the
+previous ``priceFilter()`` method::
+
+    // src/AppBundle/Twig/AppRuntime.php
+    namespace AppBundle\Twig;
+
+    class AppRuntime
+    {
+        // for complex Twig extensions you may need to define a constructor
```
Comment (fabpot) [inline templating/twig_extension.rst]: I would even say that if you don't have dependencies, you should never create a runtime.

**Review by fabpot — APPROVED** [review]

File: templating/twig_extension.rst (hunk 1 above)
Comment (javiereguiluz) [reply templating/twig_extension.rst]: Yes, it's not very common, but we add those version numbers for Twig and other apps different than Symfony in some places of the docs.

File: templating/twig_extension.rst (hunk 2 above)
Comment (javiereguiluz) [reply templating/twig_extension.rst]: I agree. I've reworded some things. Thanks!

**Pull request closed without merging**